cmd/release/github.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v25/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+type app struct {
+	owner  string
+	repo   string
+	client *github.Client
+}
+
+var (
+	contextTimeout = time.Second * 10
+)
+
+func getClient(accessToken string) *github.Client {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{
+			AccessToken: accessToken,
+		},
+	)
+
+	client := github.NewClient(oauth2.NewClient(ctx, ts))
+
+	return client
+}
+
+func new(accessToken string) *app {
+	return &app{
+		owner:  "MeilleursAgents",
+		repo:   "terraform-provider-ansiblevault",
+		client: getClient(accessToken),
+	}
+}
+
+func (a *app) PublishArtifactsOnTag(tag string, buildAssets []string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), contextTimeout)
+	defer cancel()
+
+	log.Println("==> Retrieve release informations from tag ...")
+	release, err := a.retrieveTag(ctx, tag)
+
+	if err != nil {
+		return errors.Wrapf(err, "[ERROR] When retrieve release information %s", release.GetHTMLURL())
```

Comment:
`pkg/errors` store the stack trace of error. You don't have to precise "when". It's also an error by design, it doesn't need to be precised in comment.

```suggestion
		return errors.Wrapf(err, "unable to retrieve release information for %s", release.GetHTMLURL())
```

Reply:
I've put `[ERROR]` because without it's not very visible
```
2019/07/04 09:01:54 ==> Retrieve release assets ...
2019/07/04 09:01:54 [ERROR] When retrieve assets https://github.com/MeilleursAgents/terraform-provider-ansiblevault/releases/tag/v1.0.1: GET https://api.github.com/repos/helo/terraform-provider-ansiblevault/releases/17425845/assets?page=1: 404 Not Found []
exit status 1
make: *** [github] Error 1
```

vs
```
2019/07/04 09:02:28 ==> Retrieve release assets ...
2019/07/04 09:02:28 unable to retrieve assets https://github.com/MeilleursAgents/terraform-provider-ansiblevault/releases/tag/v1.0.1: GET https://api.github.com/repos/helo/terraform-provider-ansiblevault/releases/17425845/assets?page=1: 404 Not Found []
exit status 1
make: *** [github] Error 1
```